### ISSUE TITLE
Use Sentry issue priority to specify `severity` on outgoing events

### ISSUE THREAD

**Issue by tiny-crab** [getsentry/sentry#77987]
### Problem Statement

My team uses Sentry to create incidents on Pagerduty when a new event is added to an issue. Sometimes there are noisy issues that come from error logs or thrown exceptions, and we want to be able to control the noise without outright archiving the issue.

I set up Event Orchestration in Pagerduty that sets the urgency of the incident based on the `severity` sent by Sentry. When the severity is `error`, we page the on-call. When the severity is `warning`, we file an incident but do not page the oncall.

I'd like to be able to set a `severity` level based on the Sentry Issue priority, rather than the underlying alert. This way, if a noisy issue pages the on-call, they would be able to set the Issue Priority in Sentry to `Medium`. Then they will still receive incidents, but won't be woken up in the middle of the night from a page.

![Image](https://github.com/user-attachments/assets/7d4ec2c1-6784-4a8b-96e1-f6c56c0faa15)

Sentry currently only allows me to filter based off of event level, not Issue Priority. 

![Image](https://github.com/user-attachments/assets/ebf28f58-550a-4c80-9a81-37373016e532)

### Solution Brainstorm

I'd like to be able to set a filter based off of Issue Priority. This way, I could set up an alert that declares:

1. When an event is captured by Sentry, and the Issue's Priority is High
2. Send a notification to Pagerduty with Error severity

And another alert that could declare:

1. When an event is captured by Sentry, and the Issue's Priority is Medium
2. Send a notification to Pagerduty with Warning severity.

And so on for Low priority issues.

### Product Area

Alerts

**Comment by rachrwang** [comment]
@tiny-crab - thank you for filing this feature idea! We're making some bigger changes to how alerts are configured, and filtering based on priority is on our list of changes to add in

**Comment by tiny-crab** [comment]
@rachrwang That's great to hear. I'm looking forward to seeing the new features. What's the timeline for this?

**Comment by schew2381** [comment]
Hey @tiny-crab, I think as a very rough estimate I'd say EOY or beginning of next year.

**Comment by tiny-crab** [comment]
Okay, that's good to hear. Excited for that to be built and released. How will I know when it's complete?

**Comment by schew2381** [comment]
We have some larger UI changes planned, so it should be very noticeable when it takes effect 😅

**Comment by tiny-crab** [comment]
Cool, looking forward to seeing the new changes.

**Comment by rachrwang** [comment]
@tiny-crab - our changes will start rolling out in the first half of next year. Thanks!

**Comment by ceorourke** [comment]
@tiny-crab we have added these new priority filters that might meet your needs:

<img width="1027" alt="Image" src="https://github.com/user-attachments/assets/3a2b0b66-c599-46e5-9ae6-3e08b13b0928" />

**Comment by tiny-crab** [comment]
@ceorourke I just checked those out, thanks for the tip.

What does it mean by `marks` in those two filters? Does the priority need to _change_ in order to trigger an alert using that event type?

If I'm not mistaken, I'm more looking for a filter in the `IF` section of the alert configuration. Since I want to manually change the Sentry issue priority based on noise, and have it sent at a lower severity to Pagerduty.

**Comment by saponifi3d** [comment]
> If I'm not mistaken, I'm more looking for a filter in the IF section of the alert configuration. Since I want to manually change the Sentry issue priority based on noise, and have it sent at a lower severity to PagerDuty.

Yeah, you're correct that you'd want to have the `if` section create a filter to then send that to pager duty. The `marks` info is basically saying "If an issue's priority is set or changed to high, then trigger the following actions"

---

As @rachrwang mentioned, we're actively working on a feature to make this waayyy more robust. We'll have a configuration to allow you to define the issue priority as you described. That way you could just create issues that are lower priority for your team as low priority issues. We'll also introduce the `if` filter that you described as a way to handle the severity levels. We think the combination of features will allow you to really hone into when / how to be alerted about issues.

**Comment by tiny-crab** [comment]
@saponifi3d Thanks! That's exciting to hear. @rachrwang had mentioned these features would be rolling out in H1 2025... any updates on how soon that would be? :D

**Comment by rachrwang** [comment]
@tiny-crab - we don't have any updates yet on timing, though we will definitely follow up as we have more details to share!

**Comment by ceorourke** [comment]
We're still actively working on this, it should be released in the new few months but I can't make promises regarding the timeline.
